app/models/eligibility_verification.rb

```diff
@@ -17,24 +30,38 @@ def self.expiration_warning
   end
 
   def will_expire_within_warning_period?
+    return if expiration_date.blank?
+
     expiration_date < EligibilityVerification.expiration_warning &&
       expiration_date >= Date.today
   end
 
   def expired_within_grace_period?
+    return true if expiration_date.blank?
+
     return false if expiration_date >= Time.zone.today
     expiration_date > EligibilityVerification.grace_period
   end
 
   def expired?
-    expiration_date < EligibilityVerification.grace_period
+    date = expiration_date || passenger.registration_date
+
+    date < EligibilityVerification.grace_period
   end
 
   private
 
-  def temporary_passenger
-    return if passenger.temporary?
+  def passenger_requires_validation
+    passenger&.active? && passenger&.temporary?
+  end
 
-    errors.add :base, 'must belong to a temporary passenger'
+  def wipe_contact_info
```

Comment:
This method name is a little frightening, it mostly doesn't "wipe" the contact info.

(Not that I can think of a _better_ name)